frontend/components/portfolio/ListView.vue

```diff
@@ -0,0 +1,224 @@
+<script setup lang="ts">
+import { portfolioEditor } from '@/lib/editor'
+import { type Portfolio, type PortfolioGroup } from '@/openapi/generated/pacta'
+
+const {
+  humanReadableTimeFromStandardString,
+  humanReadableDateFromStandardString,
+} = useTime()
+const pactaClient = usePACTA()
+const { loading: { withLoading } } = useModal()
+const i18n = useI18n()
+const { t } = i18n
+
+interface Props {
+  portfolios: Portfolio[]
+  portfolioGroups: PortfolioGroup[]
+  selectedPortfolioIds: string[]
+  selectedPortfolioGroupIds: string[]
+}
+const props = defineProps<Props>()
+interface Emits {
+  (e: 'update:selectedPortfolioIds', value: string[]): void
+  (e: 'update:selectedPortfolioGroupIds', value: string[]): void
+  (e: 'refresh'): void
+}
+const emit = defineEmits<Emits>()
+
+const refresh = () => { emit('refresh') }
+
+const selectedPortfolioIDs = computed({
+  get: () => props.selectedPortfolioIds ?? [],
+  set: (value: string[]) => { emit('update:selectedPortfolioIds', value) },
+})
+
+interface EditorObject extends ReturnType<typeof portfolioEditor> {
+  id: string
+}
+
+const prefix = 'components/portfolio/ListView'
+const tt = (s: string) => t(`${prefix}.${s}`)
+const expandedRows = useState<EditorObject[]>(`${prefix}.expandedRows`, () => [])
+const selectedRows = computed<EditorObject[]>({
+  get: () => {
+    const ids = selectedPortfolioIDs.value
+    return editorObjects.value.filter((editorObject) => ids.includes(editorObject.id))
+  },
+  set: (value: EditorObject[]) => {
+    const ids = value.map((row) => row.id)
+    ids.sort()
+    selectedPortfolioIDs.value = ids
+  },
+})
+
+const editorObjects = computed<EditorObject[]>(() => props.portfolios.map((item) => ({ ...portfolioEditor(item, i18n), id: item.id })))
```

Comment:
I'm not sure if this construct is guaranteed to work reliably, since `portfolioEditor` is calling `computed` itself, which will be invoked after `setup` runs. I think `computed` is one of those functions that you're only supposed to run in the `setup` function itself.

As for a fix, the `props` should already be reactive, so just calling `portfolioEditor` for each portfolio in `props.portfolio` at `setup` time _might_ be enough to make it work. Otherwise we might need to shake the paradigm up a bit.

Reply:
Ah, but we can't do that either because

`Getting a value from the props in root scope of setup() will cause the value to lose reactivity`

This does work FWIW. I think I can explain how too: when we need to access the structure (i.e. how many elements in the array), the first `.value` call will cause that to be reactive. When we need to access the content (i.e. a specific element of the array's internals) we're doing an additional `.value` call. That can just be thought of as two computed values, so while I don't think we'd get cross-ref reactivity (i.e. changing a value of an inner ref wouldn't trigger reactivity in an outer ref), that's actually OK so long as we're de-referencing both in any instance that requires access to both values (which we are). See line 112 for an example. The `editorObjects` value in the PDataTable makes the rows reactive, and the additional dereference to `editorValues.value` within the template makes the row value reactive. 

I agree this is more complicated than is ideal, and I'm open to restructuring ideas, I just can't think of how to do that in a way that allows for pure-read `computed`s and bidirectional `computed`s in the same object. And FWIW, this works without error logs, which is nice.